common/djangoapps/student/models.py

```diff
@@ -3461,6 +3463,27 @@ def should_celebrate_first_section(enrollment):
         except CourseEnrollmentCelebration.DoesNotExist:
             return False
 
+    @staticmethod
+    def should_celebrate_weekly_goal(enrollment):
+        """ Returns the celebration value for weekly_goal with appropriate fallback if it doesn't exist """
+        if not enrollment:
+            return False
+        try:
+            if enrollment.celebration.celebrate_weekly_goal:
+                # Avoiding circular import
+                from lms.djangoapps.course_goals.models import CourseGoal, UserActivity
+                goal = CourseGoal.objects.get(user=enrollment.user, course_key=enrollment.course.id)
+                today = date.today()
+                monday_date = today - timedelta(days=today.weekday())
+                week_activity_count = UserActivity.objects.filter(
+                    user=enrollment.user, course_key=enrollment.course.id, date__gte=monday_date,
+                ).count()
+                # days_per_week defaults to 0 and we don't want that to sneak in
+                return goal.days_per_week and week_activity_count == goal.days_per_week
```

Comment:
nit: could early exit on falsy `days_per_week` and avoid the user activity query?